### PR TITLE
DDF-3837 Refactoring MetacardToKml to multiple methods

### PR DESCRIPTION
**Pull request by Bdthomson**
#### What does this PR do?
Refactors MetacardToKml's `createKmlGeoFromWkt` method to do only what it's name implies and pull some inner logic into another method.

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)

@glenhein 
@brianfelix 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)

@bdeining
@coyotesqrl

#### How should this be tested? (List steps with links to updated documentation)
Full build should suffice, refactored a couple methods without changing functionality.

#### Any background context you want to provide?
Currently the `createKmlGeoFromWkt` method has side effects that are unclear until digging into the function.

#### What are the relevant tickets?
[DDF-3837](https://codice.atlassian.net/browse/DDF-3837)
#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java [hunk 1]
```diff
@@ -45,21 +45,25 @@
...
+  public static Geometry addJtsGeoPointsToKmlGeo(
+      com.vividsolutions.jts.geom.Geometry jtsGeo, Geometry kmlGeo) {
+    if (!POINT_TYPE.equals(jtsGeo.getGeometryType())) {
+      kmlGeo = addJtsCoordinateToKmlGeo(kmlGeo, jtsGeo.getCoordinate());
     }
     return kmlGeo;
   }
 
-  private static Geometry createKmlGeometry(com.vividsolutions.jts.geom.Geometry jtsGeometry)
+  public static Geometry createKmlGeoFromJtsGeo(com.vividsolutions.jts.geom.Geometry jtsGeometry)
```
Comment (coyotesqrl) [inline catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java]: ✏️ Should be able to write positive & negative tests for this newly public method now.

File: catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java [hunk 2]
```diff
@@ -111,7 +115,7 @@ private static Geometry createPointGeo(Point jtsPoint) {
     return KmlFactory.createPoint().addToCoordinates(jtsPoint.getX(), jtsPoint.getY());
   }
 
-  private static com.vividsolutions.jts.geom.Geometry readGeoFromWkt(final String wkt)
+  public static com.vividsolutions.jts.geom.Geometry createJtsGeoFromWkt(final String wkt)
```
Comment (coyotesqrl) [inline catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java]: ✏️ Should be able to write positive & negative tests for this newly public method now.

File: catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java (hunk 1 above)
Comment (Bdthomson) [reply catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java]: Added Tests

File: catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java (hunk 2 above)
Comment (Bdthomson) [reply catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java]: Added Tests

File: catalog/spatial/kml/spatial-kml-util/pom.xml [hunk 3]
```diff
@@ -84,12 +84,12 @@
                                         <limit>
                                             <counter>INSTRUCTION</counter>
                                             <value>COVEREDRATIO</value>
-                                            <minimum>0.76</minimum>
+                                            <minimum>0.80</minimum>
```
Comment (glenhein) [inline catalog/spatial/kml/spatial-kml-util/pom.xml]: Does this include a 5% buffer?

File: catalog/spatial/kml/spatial-kml-util/pom.xml [hunk 4]
```diff
@@ -84,12 +84,12 @@
...
                                         </limit>
                                         <limit>
                                             <counter>BRANCH</counter>
                                             <value>COVEREDRATIO</value>
-                                            <minimum>0.73</minimum>
+                                            <minimum>0.75</minimum>
```
Comment (glenhein) [inline catalog/spatial/kml/spatial-kml-util/pom.xml]: Does this include a 5% buffer?

File: catalog/spatial/kml/spatial-kml-util/pom.xml (hunk 3 above)
Comment (glenhein) [reply catalog/spatial/kml/spatial-kml-util/pom.xml]: I would include a comment that indicates a 5% buffer is included

**Comment by Bdthomson** [comment]
build now

File: catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java [hunk 5]
```diff
@@ -111,16 +115,22 @@ private static Geometry createPointGeo(Point jtsPoint) {
     return KmlFactory.createPoint().addToCoordinates(jtsPoint.getX(), jtsPoint.getY());
   }
 
-  private static com.vividsolutions.jts.geom.Geometry readGeoFromWkt(final String wkt)
+  public static com.vividsolutions.jts.geom.Geometry getJtsGeoFromWkt(final String wkt)
       throws CatalogTransformerException {
+
+    if (StringUtils.isBlank(wkt)) {
+      throw new CatalogTransformerException(
+          "WKT was null or empty. Unable to preform KML Transform on Metacard.");
```
Comment (Bdthomson) [inline catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java]: This probably needs to change to "WKT was null or empty. Unable to convert WKT to Jts Geometry."

File: catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java [hunk 6]
```diff
@@ -111,16 +115,22 @@ private static Geometry createPointGeo(Point jtsPoint) {
     return KmlFactory.createPoint().addToCoordinates(jtsPoint.getX(), jtsPoint.getY());
   }
 
-  private static com.vividsolutions.jts.geom.Geometry readGeoFromWkt(final String wkt)
+  public static com.vividsolutions.jts.geom.Geometry getJtsGeoFromWkt(final String wkt)
```
Comment (paouelle) [inline catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java]: ✏️ if `wkt` can be `null` then it should be annotated with `@Nullable`. If not then we can replace the `StringUtils.isBlank(wkt)` below with `wkt.isEmpty()`

**Comment by Bdthomson** [comment]
build now

**Pull request merged**